collection-scripts/gather_services_status

```diff
@@ -115,6 +118,25 @@ get_placement_status() {
     run_bg ${BASH_ALIASES[os]} trait list '>' "$PLACEMENT_PATH"/trait_list
 }
 
+# Ironic service gathering - nodes, ports, conductors if we can get them.
+get_ironic_status() {
+    local IRONIC_PATH="$BASE_COLLECTION_PATH/ctlplane/ironic"
+    mkdir -p "$IRONIC_PATH"
+    # NOTE(TheJulia): The idea here is to try and collect information visible,
+    # as Ironic has filtering in place on all project scoped requests,
+    # as agreed by the Ironic community.
+    run_bg ${BASH_ALIASES[os]} baremetal node list --long '>' "$IRONIC_PATH"/node_list
+    run_bg ${BASH_ALIASES[os]} baremetal port list --long '>' "$IRONIC_PATH"/port_list
+    run_bg ${BASH_ALIASES[os]} baremetal portgroup list --long '>' "$IRONIC_PATH"/portgroup_list
+    run_bg ${BASH_ALIASES[os]} baremetal volume connector list --long '>' "$IRONIC_PATH"/volume_connector_list
+    run_bg ${BASH_ALIASES[os]} baremetal volume target list --long '>' "$IRONIC_PATH"/volume_target_list
+    run_bg ${BASH_ALIASES[os]} baremetal allocation list --long '>' "$IRONIC_PATH"/allocation_list
+    # Driver/Conductor lists are restricted endpoints by default since
+    # they provide insight into the overall infrastucture which would
+    # be inappropriate in a public cloud context and we don't inherently
+    # have the required elevated rights "out of the" box to gather it.
```

Comment:
I suppose we can't also collect all inspection and deploy logs?

Reply:
I don't think so. we would have to reach into containers *or* sidecar it all. I kind of like metal3's approach on this regard to just stream the console logging.

Reply:
Steve and I discussed this earlier today, and we agree we should try and collect that data, since it looks like metal3's approach is not viable for our existing setup. That being said, I think we're going to want to go ahead and handle it under another issue and as a result a separate pull request because it will take a little more pondering to achieve.

Reply:
Sounds good. Feel free to open a different patch for any follow up work!